### PR TITLE
MKL-DNN:  fix batchnorm unit test failures

### PR DESCRIPTION
**Pull request by gzmkl**
Fix failures of all 9 fuse batchnorm test cases.

handle corner case (empty input tensors)
handle inference case properly - bwd bug related to fwd primitive creation (as a hint)
refactor - moving output tensor allocation to separate methods - to avoid duplicated code

### CODE REVIEW

File: tensorflow/core/kernels/mkl_fused_batch_norm_op.cc [hunk 1]
```diff
@@ -703,27 +703,31 @@ class MklFusedBatchNormOp : public OpKernel {
...
-      const size_t mean_index = 3;   // index of est_mean tensor
-      const size_t var_index = 4;    // index of est_variance tensor
-
-      const Tensor& src_tensor          = MklGetInput(context, src_index);
-      const Tensor& scale_tensor        = MklGetInput(context, scale_index);
-      const Tensor& shift_tensor        = MklGetInput(context, shift_index);
-      const Tensor& est_mean_tensor     = MklGetInput(context, mean_index);
-      const Tensor& est_variance_tensor = MklGetInput(context, var_index);
-
+      const size_t kSrcIndex = 0;       // index of src input tensor
```
Comment (qmick) [inline tensorflow/core/kernels/mkl_fused_batch_norm_op.cc]: Just curious, why rename the constants of type `size_t` while keeping Tensors unchanged? In the code guideline, such naming convention is optional for local non-static variables.

**Review by andydavis1 — APPROVED** [review]

**Comment by andydavis1** [comment]
@tensorflow-jenkins test this please

File: tensorflow/core/kernels/mkl_fused_batch_norm_op.cc (hunk 1 above)
Comment (gunan) [reply tensorflow/core/kernels/mkl_fused_batch_norm_op.cc]: To me this makes sense, as any variable that starts with "k" is a constant. And here, they are really constants, their values will never ever change. Almost like macros.
When I look at the tensors below, they are constants but their values will be set at runtime.

So IMHO this makes the code more readable for me.

**Review by gunan — APPROVED** [review]
